dropshot/src/websocket.rs

```diff
@@ -184,45 +172,43 @@ impl Extractor for WebsocketUpgrade {
-    * used with your choice of websocket-handling code operating over an
-    * [`tokio::io::AsyncRead`] + [`tokio::io::AsyncWrite`] type
-    * (e.g. `tokio_tungstenite`).
-    *
-    * ```
-      #[dropshot::endpoint { method = GET, path = "/my/ws/endpoint/{id}" }]
-      async fn my_ws_endpoint(
-          rqctx: std::sync::Arc<dropshot::RequestContext<()>>,
-          websock: dropshot::WebsocketUpgrade,
-          id: dropshot::Path<String>,
-      ) -> dropshot::WebsocketEndpointResult {
-          let logger = rqctx.log.new(slog::o!());
-          websock.handle(move |upgraded| async move {
-              slog::info!(logger, "Entered handler for ID {}", id.into_inner());
-              use futures::stream::StreamExt;
-              let mut ws_stream = tokio_tungstenite::WebSocketStream::from_raw_socket(
-                  upgraded.into_inner(), tokio_tungstenite::tungstenite::protocol::Role::Server, None
-              ).await;
-              slog::info!(logger, "Received from websocket: {:?}", ws_stream.next().await);
-              Ok(())
-          })
-      }
-    * ```
-    *
-    * Note that as a consumer of this crate, you most likely do not want to
-    * call this function directly; rather, prefer to annotate your function
-    * with [`dropshot_endpoint::channel`] instead of `endpoint`.
-    */
+    /// Upgrade the HTTP connection to a websocket and spawn a user-provided
+    /// async handler to service it.
+    ///
+    /// This function's return value should be the basis of the return value of
+    /// your endpoint's function, as it sends the headers to tell the HTTP
+    /// client that we are accepting the upgrade.
+    ///
+    /// `handler` is a closure that accepts a [`WebsocketConnection`]
+    /// and returns a future that will be spawned by this function,
+    /// in which the `WebsocketConnection`'s inner `Upgraded` connection may be
+    /// used with your choice of websocket-handling code operating over an
+    /// [`tokio::io::AsyncRead`] + [`tokio::io::AsyncWrite`] type
+    /// (e.g. `tokio_tungstenite`).
+    ///
+    /// ```
+    /// #[dropshot::endpoint { method = GET, path = "/my/ws/endpoint/{id}" }]
+    /// async fn my_ws_endpoint(
+    /// rqctx: std::sync::Arc<dropshot::RequestContext<()>>,
+    /// websock: dropshot::WebsocketUpgrade,
+    /// id: dropshot::Path<String>,
+    /// ) -> dropshot::WebsocketEndpointResult {
+    /// let logger = rqctx.log.new(slog::o!());
+    /// websock.handle(move |upgraded| async move {
+    /// slog::info!(logger, "Entered handler for ID {}", id.into_inner());
+    /// use futures::stream::StreamExt;
+    /// let mut ws_stream = tokio_tungstenite::WebSocketStream::from_raw_socket(
+    /// upgraded.into_inner(), tokio_tungstenite::tungstenite::protocol::Role::Server, None
+    /// ).await;
+    /// slog::info!(logger, "Received from websocket: {:?}", ws_stream.next().await);
+    /// Ok(())
+    /// })
+    /// }
```

Comment:
lost the formatting

Reply:
Great catch!  For my future reference, I think it's because it wasn't formatted the way standard block-style comments are formatted (missing leading asterisks on those lines).  I fixed this by reverting this file, fixing the C-style formatting, then rerunning `cargo +nightly fmt -- --config-path=rustfmt-new.toml`.